_minutes/2022-02-17-wecg.md

```diff
@@ -0,0 +1,134 @@
+ * [simeon] Our MV3 docs are missing CSP documentation. It's on my backlog.
+ * [rob] Rather than creating a new one on the Chrome website, how about linking to MDN?
+ * [simeon] That's a more complicated issue worthy of it's own topic. For now let's continue to focus on CSP.
+ * [simeon] Sounds like we need to gather more information. Suggest that we gather more information in issue 98 and follow up next session.
+ * [carlos] Chrome used to require a CSP exception for WebAssembly, but a recent update to Chrome made it optional with a Chrome flag.
+ * [bradley] With the recent Chrome update, the wasm-eval directive can now be specified to allow WASM execution in extensions for MV3. After the update however, there is no longer the need to specify any CSP for WASM to work properly whether the extension is run as MV2 or MV3. WASM being used is locally packaged in the extension.
+
+[Issue 96](https://github.com/w3c/webextensions/issues/96): Inconsistency: content_security_policy syntax
+
+ * [rob] Object syntax is the only syntax going forward in MV3 extensions. Object syntax is currently only supported in MV3; the string version is MV2-only, and we may support the object syntax in MV2 in Firefox to help with transitioning.
+ * [timothy] Since there's a manifest version change, I think it makes the most sense to expect developers to update to the new syntax. Polymorphism is not great.
+
+[Issue 95](https://github.com/w3c/webextensions/issues/95): Proposal: content_security_policy.content_scripts
+
+ * [simeon] We intend to provide a mechanism to allow extensions to run code in the main world via the scripting API, mainly for userScripts.
+ * [alexei] In MV2 it's possible to run code in the main world from the content script before anything happens. This doesn't work in MV3 because of the CSP.
+ * [rob] Adding a method to the scripting API requires the extension to call the method from the background script, it's not available in content scripts. https://bugs.chromium.org/p/chromium/issues/detail?id=1207006
+ * [jack] if arbitrary code is not allowed in MV3, how can Tampermonkey run user scripts?
+ * [simeon] The scripting API will be expanded to support user scripts. (post meeting edit: we to gate access to this capability similar to Chrome's current "allow access to file URLs" control)
+ * [tomislav] Two features are being mixed up in this discussion: 1) running scripts in the main world synchronously and 2) running arbitrary scripts (not shipped with extension) in the main world from a content script.
+ * [simeon] The world property of the scripting API allows the selection of the world. We believe that this allows extensions to schedule scripts in the desired order.
```

Comment:
```suggestion
 * [simeon] The world property of the scripting API allows the selection of the world. We believe that this allows extensions to schedule scripts in the desired order.
 * [simeon] Let's try a different framing. In order to inject a content script into the main world, an extension developers would declare a content script object in manifest.json, but add a new "world" property. This prop takes a value of either MAIN or ISOLATED. The scripting API's dynamic content script support will also be expanded to support this property.
```

Reply:
I've added this line to the notes, but I note that this comment is part of a larger conversation during the meeting where I and other participants tried to emphasize that there are use cases where extensions want to decide at runtime where main world scripts are injected by the content script. The "func" feature of `scripting.executeScript` is to allow extensions to run logic without being forced to run create separate files for every bit of logic.

Neither dynamic nor static (manifest) content scripts would allow extensions to synchronously run main world scripts from the content script.